main_test.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+)
+
+var a Job
+
+func TestMain(m *testing.M) {
+
+	a.Initialize("DB_USERNAME", "DB_PASSWORD", "rest_api_example")
+	ensureTableExists()
+	code := m.Run()
+	clearTable()
+	os.Exit(code)
+}
+func ensureTableExists() {
+	if _, err := a.DB.Exec(tableCreationQuery); err != nil {
+		log.Fatal(err)
+	}
+}
+func clearTable() {
+	a.DB.Exec("DELETE FROM job")
+	a.DB.Exec("ALTER TABLE job AUTO_INCREMENT = 1")
+}
+
+const tableCreationQuery = `
+CREATE TABLE IF NOT EXISTS job
+(
+    id INT AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(50) NOT NULL
+)`
+
+func TestEmptyTable(t *testing.T) {
+	clearTable()
+	req, _ := http.NewRequest("GET", "/api/job", nil)
+	response := executeRequest(req)
+	checkResponseCode(t, http.StatusOK, response.Code)
+	if body := response.Body.String(); body != "[]" {
```

Comment:
instead of transforming this on a `String()` you wanna compare the Body structure directly.

Reply:
https://blog.questionable.services/article/testing-http-handlers-go/